include/util/enum.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace waybar::util {
+
+struct EnumParser {
+  EnumParser() {}
+
+  enum SORT_METHOD { ID, NAME, NUMBER, DEFAULT };
+
+  SORT_METHOD sortStringToEnum(const std::string& str) {
+    // Convert the input string to uppercase (make it lenient on config input)
+    std::string uppercaseStr;
+    for (char c : str) {
+        uppercaseStr += std::toupper(c);
+    }
+
+    static const std::map<std::string, SORT_METHOD> enumMap = {
+        {"ID", ID}, {"NAME", NAME}, {"NUMBER", NUMBER}, {"DEFAULT", DEFAULT}};
```

Comment:
The problem is also that you're defining a fixed map here: if you want to add a way for other modules to parse enums, you'd have to pass the enum map to the EnumParser constructor.

Reply:
I mean, this was specific to sort-by. If you want to take a stab at creating a more robust enum system I can just back out those changes and you can have at it.

Reply:
Here's what I propose: https://github.com/Alexays/Waybar/commit/c628bc780c53c6e5c90cefcaf427cdcd4ab438b9

I'd love to look into a more robust enum parsing system, but indeed better in a different PR :)

Reply:
I might have something that works like you are mentioning, though. I'll take one last stab at it and if we want to change it we can refine it :P